File: C#/AoC/AoC2024/dec1/1.cs
namespace AoC2024.dec1;

public class Dec1
{
    public static void Run()
    {
        // Read and parse input, skipping empty or invalid lines
        var lines = File.ReadAllLines("../../../dec1/input.txt")
            .Where(line => !string.IsNullOrWhiteSpace(line))
            .Select(line => line.Split().Where(part => int.TryParse(part, out _)).Select(int.Parse).ToArray())
            .Where(parts => parts.Length == 2)
            .ToArray();

        // Separate and sort values
        var leftList = lines.Select(x => x[0]).OrderBy(x => x).ToArray();
        var rightList = lines.Select(x => x[1]).OrderBy(x => x).ToArray();

        // Calculate and print total distance
        var totalDistance = leftList.Zip(rightList, (l, r) => Math.Abs(l - r)).Sum();
        Console.WriteLine(totalDistance);
    }
}

File: Python/aoc/2024/dec1/1.py
import os

# read the input file
with open(os.path.join(os.path.dirname(__file__), 'input.txt')) as f:
    lines = f.readlines()

# ---------------------------------------------------------------- #

# parse and sort the input into two lists
left_list, right_list = [], []
for line in lines:
    left, right = map(int, line.split())
    left_list.append(left)
    right_list.append(right)

# sort both lists
left_list.sort()
right_list.sort()

# calculate and print the total distance
print(sum(abs(left - right) for left, right in zip(left_list, right_list)))  # 1941353